examples/create_managed_spu.ts

```diff
@@ -0,0 +1,39 @@
+import Fluvio, { Encryption, CustomSpuSpec } from '../src/index'
+
+console.log('connecting client to sc')
+
+async function createManagedSpu() {
+    try {
+        const fluvio = new Fluvio({
+            host: '10.97.193.189',
+            port: 9003,
+            minId: 10,
+            replicas: 2,
+        })
+
+        // Explicitly call `.connect()` to connect to the cluster;
+        // This allows for lazily-loading the connection, useful in
+        // situations where the fluvio client does not need to immediately
+        // connect.
+        await fluvio.connect()
+
+        // Set the admin client;
+        await fluvio.setAdmin()
```

Comment:
I'm not a fan of this client object having so many stateful options, it feels like it'd be pretty easy to misuse the API unintentionally. For example, it seems that any "admin" functionality would be invalid to invoke until after calling `setAdmin`, is that correct? If that's the case, then the API inadvertently makes invalid flows possible. Let's take this `createManagedSpu` functionality as an example. With the current API, this flow is possible, but not valid:

```Typescript
const fluvio = new Fluvio({ ... });
await fluvio.connect();
// Forget to call fluvio.setAdmin()
fluvio.createManagedSpu( ... );
```

In the Rust client, API functionality is gated behind the constructor for each type of client object. There are two significant gates:

1) You must successfully connect (and authenticate, if TLS is enabled) to Fluvio before a `Fluvio` client object may be created
2) You must be authorized as an administrator in order to construct a `FluvioAdmin` client object

Since portions of the API are attached to those object types as methods, it is simply impossible to call those methods while in an invalid state, because by virtue of having those objects you are in the correct state. I think a similar pattern should be applied to the Node client, though the specific mechanics of that API-gating may look slightly different. I think that one easy step would be to break the admin functionality into its own object. That way there wouldn't be a `setAdmin` method to mutate the state of one overloaded Fluvio client. Instead, there might be an `admin(): FluvioAdmin?` method that constructs the admin client only if authorized.